src/hotspot/share/compiler/compileTask.cpp

```diff
@@ -133,6 +76,33 @@ void CompileTask::initialize(int compile_id,
   _arena_bytes = 0;
 
   _next = nullptr;
+
+  Atomic::add(&_active_tasks, 1);
+}
+
+CompileTask::~CompileTask() {
+  if ((_method_holder != nullptr && JNIHandles::is_weak_global_handle(_method_holder))) {
```

Comment:
While moving the code, you can probably remove one pair of parentheses here:
```suggestion
  if (_method_holder != nullptr && JNIHandles::is_weak_global_handle(_method_holder)) {
```